packages/website/src/components/Playground.tsx

```diff
@@ -171,19 +158,34 @@ function Playground(): JSX.Element {
                   />
                 )}
               </div>
-
               {(state.showAST === 'es' && esQueryError && (
                 <ErrorViewer
                   type="warning"
                   title="Invalid Selector"
                   value={esQueryError}
                 />
               )) ||
-                (state.showAST && astToShow && (
+                (state.showAST === 'types' && astModel?.storedTsAST && (
+                  <TypesDetails
+                    typeChecker={astModel.typeChecker}
+                    value={astModel.storedTsAST}
+                    onHoverNode={setSelectedRange}
+                    cursorPosition={position}
+                  />
+                )) ||
+                (state.showAST && astModel && (
```

Comment:
i have a feeling that i should extract this to separate component, this and or condition is getting oo hands :P

Reply:
yeah this is probably getting to the point where we might want to extract the condition into a bit of state machine logic which outputs an enum value so that we can simplify the render logic.

Reply:
Agreed. I don't personally think of it as a blocker but it would be a nice to have. Maybe a good follow-up `good first issue` for a potential contributor?

Reply:
Filed #7630. On second thought I think it's probably not a `good first issue` given it's touching a bunch of code.